app/src/main/java/rick_and_morty/data/repository/CharacterRepository.kt

```diff
@@ -0,0 +1,24 @@
+package rick_and_morty.data.repository
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import rick_and_morty.di.RickAndMortyApiRemoteDataSource
+import javax.inject.Inject
+import rick_and_morty.data.model.CharacterResultsDto
+
+
+class CharacterRepository @Inject constructor(
+    private val rickAndMortyApiRemoteDataSource: RickAndMortyApiRemoteDataSource
+){
+    
+   fun getCharacters(page: Int): Flow<List<CharacterResultsDto>> = flow {
+        emit(rickAndMortyApiRemoteDataSource.fetchRickAndMortyData(page).results!!)
+    }.flowOn(Dispatchers.Default)
```

Comment:
`Retrofit` is already using `Dispatchers.IO` per default. No manual switching the dispatcher is needed since we are only making the API call here.

Reply:
Done